### DISCUSSION TITLE
Unable to dismiss LNPopup Controller using button

### DISCUSSION THREAD

**Discussion by dfeler** [LeoNatan/LNPopupController#499] [General] [answered]
I am able to successfully present the bar and dismiss the bar itself when on ViewController1 - but I am struggling to dismiss the screen that opens

**For example on `ViewController1` I present `ViewController2` by doing the following:**

    let ViewController2 = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "ViewController2") as! ViewController2
            let navigationController = UINavigationController(rootViewController: ViewController2)
            navigationController.popupItem.title = "Test"
            navigationController.popupItem.subtitle = "Test"
            let popupNextButtonItem = UIBarButtonItem(image: UIImage(named: ""), style: .plain, target: self, action: #selector(openTab))
            navigationController.popupItem.trailingBarButtonItems = [popupNextButtonItem]
            navigationController.popupBar.backgroundColor = .white
            navigationController.popupBar.inheritsAppearanceFromDockingView = false
            navigationController.popupBar.tintColor = .white
            navigationController.popupContentView.isTranslucent = false
            navigationController.popupContentView.isOpaque = true
           
            //Open popup
            tabBarController?.presentPopupBar(withContentViewController: navigationController, animated: true, completion: nil)

**This creates the popup bar itself and opens the popup `ViewController2`. Now to dismiss the view controller I am attempting to do:** 

    @IBAction func dismissButton(_ sender: Any) {
            print("Dismiss Selected") 
            tabBarController?.dismissPopupBar(animated: true, completion: nil)
    }

According to the documentation 

> If the popup is open when dismissing the popup bar, the popup content
> will also be dismissed.

Oddly this is not doing anything. "Dismiss Selected" is of course printing  but the view controller is not dismissing.

**Comment by iDevelopper** [comment, marked as answer]
You use the navigation controller to configure your popup bar. So you have to present the popup bar on the navigation controller, not on the tab bar controller.

**Comment by dfeler** [comment]
Thank you @iDevelopper I have posted one other question regarding the library I would really appreciate your guidance on that one as well. [Link to Question](https://github.com/LeoNatan/LNPopupController/issues/502)

**Reply by LeoNatan** [reply]
Use discussions instead of issues. It literally tells you to do so in the template. Also, use comments when answering a discussion reply.

**Comment by trial** [comment]
I have the same problem

```
 let vc = PopupViewController()
 tabBarController?.popupBar.progressViewStyle = .top
tabBarController?.popupBar.progressView.tintColor = .red
vc.popupItem.title = item.name
tabBarController?.presentPopupBar(withContentViewController: vc, animated: true, completion: nil)
```
the tabBarController is NOT nil, but in PopupViewController(), the tabBarController is nil, and can't dismiss PopupViewController

```
    @objc func close() {
       tabBarController?.dismissPopupBar(animated: true, completion: nil)
    }
```

**Reply by iDevelopper** [reply]
You have to use the popupPresentationContainerViewController property to retrieve it. See "UIViewController+LNPopupSupport.h".
